runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/util/Convenience.kt

```diff
@@ -35,3 +35,25 @@ public fun truthiness(value: Any?): Boolean = when (value) {
     null -> false
     else -> true
 }
+
+@InternalApi
+public fun List<Number>.max(): Double? =
+    @Suppress("UNCHECKED_CAST")
+    (this as List<Double>).maxOrNull()
+
+@InternalApi
+public fun List<String>.max(): String? = this.maxOrNull()
+
+@InternalApi
+public fun List<Number>.min(): Double? =
+    @Suppress("UNCHECKED_CAST")
+    (this as List<Double>).minOrNull()
+
+@InternalApi
+public fun List<String>.min(): String? = this.minOrNull()
+
+@InternalApi
+public fun Number.toNumber(): Double = this.toDouble()
+
+@InternalApi
+public fun String.toNumber(): Double? = this.toDoubleOrNull()
```

Comment:
**Question**: Same as above, why is `Double` always right here?

Reply:
So other parts of the code require a specific type of number and I chose double because I think all the number types we deal with can be represented as a Double (as long as they are small enough). I think I can re-write this though

Reply:
There's also the question of getting a number as a string, what type of number to choose for it